diff --git a/src/gui_w32.c b/src/gui_w32.c
index 78f252a..e7c6b51 100644
--- a/src/gui_w32.c
+++ b/src/gui_w32.c
@@ -8487,17 +8487,28 @@ tabline_wndproc(
 		}
 		break;
 	    }
+	case WM_MBUTTONDOWN:
+	    {
+		s_pt.x = GET_X_LPARAM(lParam);
+		s_pt.y = GET_Y_LPARAM(lParam);
+		if (GetTabFromPoint(hwnd, s_pt) != NULL)
+		    SetCapture(hwnd);
+		break;
+	    }
 	case WM_MBUTTONUP:
 	    {
-		TCHITTESTINFO htinfo;
-
-		htinfo.pt.x = GET_X_LPARAM(lParam);
-		htinfo.pt.y = GET_Y_LPARAM(lParam);
-		idx0 = TabCtrl_HitTest(hwnd, &htinfo);
-		if (idx0 != -1)
+		if (GetCapture() == hwnd)
 		{
-		    idx0 += 1;
-		    send_tabline_menu_event(idx0, TABLINE_MENU_CLOSE);
+		    ReleaseCapture();
+		    /* when the mouse events was executed on the same tab */
+		    pt.x = GET_X_LPARAM(lParam);
+		    pt.y = GET_Y_LPARAM(lParam);
+		    tp = GetTabFromPoint(hwnd, pt);
+		    if (tp == GetTabFromPoint(hwnd, s_pt))
+		    {
+			idx0 = tabpage_index(tp);
+			send_tabline_menu_event(idx0, TABLINE_MENU_CLOSE);
+		    }
 		}
 		break;
 	    }
