diff --git a/src/gui_w32.c b/src/gui_w32.c
index 3ade631..23c5fb5 100644
--- a/src/gui_w32.c
+++ b/src/gui_w32.c
@@ -8022,6 +8022,7 @@ GetTabFromPoint(
 
 static POINT	    s_pt = {0, 0};
 static HCURSOR      s_hCursor = NULL;
+static tabpage_T    *s_tp = NULL;
 
     static LRESULT CALLBACK
 tabline_wndproc(
@@ -8094,17 +8095,29 @@ tabline_wndproc(
 		}
 		break;
 	    }
+	case WM_MBUTTONDOWN:
+	    {
+		pt.x = GET_X_LPARAM(lParam);
+		pt.y = GET_Y_LPARAM(lParam);
+		s_tp = GetTabFromPoint(hwnd, pt);
+		if (s_tp != NULL)
+		    SetCapture(hwnd);
+		break;
+	    }
 	case WM_MBUTTONUP:
 	    {
-		TCHITTESTINFO htinfo;
-
-		htinfo.pt.x = GET_X_LPARAM(lParam);
-		htinfo.pt.y = GET_Y_LPARAM(lParam);
-		idx0 = TabCtrl_HitTest(hwnd, &htinfo);
-		if (idx0 != -1)
+		if (GetCapture() == hwnd)
 		{
-		    idx0 += 1;
-		    send_tabline_menu_event(idx0, TABLINE_MENU_CLOSE);
+		    ReleaseCapture();
+		    /* when the mouse events was executed on the same tab */
+		    pt.x = GET_X_LPARAM(lParam);
+		    pt.y = GET_Y_LPARAM(lParam);
+		    if (GetTabFromPoint(hwnd, pt) == s_tp)
+		    {
+			idx0 = tabpage_index(s_tp);
+			send_tabline_menu_event(idx0, TABLINE_MENU_CLOSE);
+		    }
+		    s_tp = NULL;
 		}
 		break;
 	    }
